en/guides/codablerouting.md

```diff
@@ -15,172 +15,288 @@ redirect_from:
-    guard let todo = todo else {
-        print("Error reading todo from Kitura, no error and no response")
-        return
-    }
-    print("Read \(todo.title) from Kitura server")
+We use a dictionary to handle cases where you may want to work with a single `User`, for example a `GET` request for a single user. 
+
+We now also need to update our `postHandler` to add the `User` we post to the `userStore`. We can do this like so:  
+```swift
+func postHandler(user: User, completion: (User?, RequestError?) -> Void ) {
+    userStore[user.name] = user
+    completion(userStore[user.name], nil)
 }
 ```
+All we're doing here is using the user's name as the key, and then the actual `User` as the value.  
+So now that we have a way of storing posted users, we can add a way for 'getting' all of them. First we need to register a `GET` endpoint like we did for `POST`. So underneath the line:  
+```swift
+router.post("/users", handler: postHandler)
+```
+Add the following:  
+```swift
+router.get("/users", handler: getAllHandler)
+```
+Once again this won't compile as we haven't defined a `getAllHandler` function yet, so lets do that now.  
+Underneath the `postHandler` fuction add the following:  
+```swift
+func getAllHandler(completion: ([User]?, RequestError?) -> Void ) {
+    let users: [User] = self.userStore.map({ $0.value })
+    completion(users, nil)
+}
+```
+So unlike the `postHandler` we don't need to pass a `User` as a parameter here as we're not wanting to send anything to the server. We still have the completion closure though, this time accepting an array of `User`. This makes sense as we're 'getting' all users, and this could be one user or it could be many users.  
+We need to convert our `userStore` dictionary to an array, a simple way of doing that is using the [map](https://developer.apple.com/documentation/swift/array/2908681-map) function Swift provides.  
+Once we have the array we can pass it to the completion, and thats it.  
+We've implemented our second route!  
+
+Like before we'll use `curl` to test that our routes work correctly (or use your preferred tool).  
+Start the server by running your XCode project and then open a terminal and run the following:  
+```
+curl -X GET \
+  http://localhost:8080/users \
+  -H 'content-type: application/json'
+```
+In the terminal you should see the following: 
+```
+[]%
+```
+It's an empty array because we haven't posted any data to the server. Lets post some data to the server:  
+```
+curl -X POST \
+  http://localhost:8080/users \
+  -H 'content-type: application/json' \
+  -d '{
+    "name": "Hello World!",
+    "age": 99
+}'
+```
+Now that the server has data, lets try running the `GET` request again:  
+```
+curl -X GET \
+  http://localhost:8080/users \
+  -H 'content-type: application/json'
+```
+You should now see:  
+```
+[{"name":"Hello World!","age":99}]%
+```
+
+We have now posted data to a server, and then been able to retrieve it without having to write much code!  
```

Comment:
I think you need to make this a punchier ending... my brain is fading but try working around something more like....
"This guide demonstrates just how easy it is with Kitura's Codable routing to get and post data to a server."
... you may want to work lines of code in too but I'll leave that to you.